YAML Mark parsing wrong when not all characters in endMark

[This piece of code](https://github.com/unbroken-dome/gradle-helm-plugin/blob/master/src/main/kotlin/org/unbrokendome/gradle/plugins/helm/util/YamlTransformingReader.kt#L228) makes the assumption that the startMark pointer will always be 0.  However, that is not always the case (see below).  In some cases the offset/pointer will be non-zero which causes a StringIndexOutOfBounds when parsing.

```
Caused by: java.lang.StringIndexOutOfBoundsException: offset 1027, count 1031, length 1031
        at org.unbrokendome.gradle.plugins.helm.util.YamlTransformingReaderKt.readFromBuffer(YamlTransformingReader.kt:228)
        at org.unbrokendome.gradle.plugins.helm.util.YamlTransformingReaderKt.access$readFromBuffer(YamlTransformingReader.kt:1)
        at org.unbrokendome.gradle.plugins.helm.util.AbstractYamlTransformingReader$ValueState.handleValueEvent(YamlTransformingReader.kt:108)
        at org.unbrokendome.gradle.plugins.helm.util.AbstractYamlTransformingReader$MappingValueState.handleEvent(YamlTransformingReader.kt:202)
        at org.unbrokendome.gradle.plugins.helm.util.AbstractYamlTransformingReader$delegate$2.invoke(YamlTransformingReader.kt:82)
```

The referenced piece of code should be changed from:
```kotlin
            // Start with the codepoints from startMark's buffer
            // Note that if startMark.pointer is non-zero this causes an out-of-bounds exception
            // Contructor is: String(buffer, offset into buffer, length from offset)
            val startPart = String(startMark.buffer, startMark.pointer, startMark.buffer.size)

            val codePointsLeft = length - (startMark.buffer.size - startMark.pointer)
```

To something like:
```kotlin
            // Start with the codepoints from startMark's buffer
            val bufferPresentInStartMark = startMark.buffer.size - startMark.pointer
            val startPart = String(startMark.buffer, startMark.pointer, bufferPresentInStartMark)

            val codePointsLeft = length - bufferPresentInStartMark
```

Which fixes the above error by not trying to parse past the end of the buffer.

--- a/src/main/kotlin/org/unbrokendome/gradle/plugins/helm/util/YamlTransformingReader.kt
+++ b/src/main/kotlin/org/unbrokendome/gradle/plugins/helm/util/YamlTransformingReader.kt
@@ -225,9 +225,10 @@ private fun readFromBuffer(startMark: Mark, endMark: Mark): String {
 
         else -> {
             // Start with the codepoints from startMark's buffer
-            val startPart = String(startMark.buffer, startMark.pointer, startMark.buffer.size)
+            val startPartLength = startMark.buffer.size - startMark.pointer
+            val startPart = String(startMark.buffer, startMark.pointer, startPartLength)
 
-            val codePointsLeft = length - (startMark.buffer.size - startMark.pointer)
+            val codePointsLeft = length - startPartLength
             if (endMark.pointer < codePointsLeft) {
                 throw IllegalStateException("Buffer unavailable")
             }

